apps/dh/api-dh/source/DataHub.WebApi/Controllers/WholesaleSettlementReportController.cs

```diff
@@ -48,10 +49,14 @@ public WholesaleSettlementReportController(
     [AllowAnonymous]
     public async Task<ActionResult<Stream>> DownloadReportAsync([FromQuery] string settlementReportId, [FromQuery] Guid token, [FromQuery] string filename, [FromQuery] bool fromApi)
     {
-        var apiClientSettings = _configuration.GetSection("ApiClientSettings").Get<ApiClientSettings>() ?? new ApiClientSettings();
-        var baseUri = GetBaseUri(apiClientSettings.WholesaleOrchestrationSettlementReportsBaseUrl);
-        var lightBaseUri = GetBaseUri(apiClientSettings.WholesaleOrchestrationSettlementReportsLightBaseUrl);
-        var apiClientBaseUri = GetBaseUri(apiClientSettings.SettlementReportsAPIBaseUrl);
+        // TODO: Change this to IOptions pattern.
+        var subSystemBaseUrls = _configuration
```

Comment:
Here you should be able to just inject IOptions<SubSystemBaseUrls>, instead of IConfiguration.

Reply:
Done